pandas/tests/frame/test_constructors.py

```diff
@@ -723,6 +724,12 @@ def test_constructor_period(self):
         assert df["a"].dtype == a.dtype
         assert df["b"].dtype == b.dtype
 
+        data = pd.Period("2012-01", freq="M")
```

Comment:
can you add the issue number and make a separate test. Also use a full comparison, e.g. assert_frame_equal here

Reply:
Writing out the full expected and using `tm.assert_frame_equal` will fit better with our other tests, and helps prevent regressions in future refactors.

Reply:
Gotcha, done!

Reply:
can you parameterize on all scalar types (as you did in the setitem tests).